### PR TITLE
Apis and Debugging

### PR DESCRIPTION
**Pull request by rvanderheyde**
I edited the content of lesson six, the former api class, instead of lesson seven, so a folder rename will be needed. I left the word cloud app in and added a passport example and an express-session example.

### CODE REVIEW

File: lessons/06-apis-debugging/INCLASS.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#Class 6
+##Facebook Design Activity
+You are designing Facebook's REST API. Come up with a set of routes to do everything Facebook does. When done, read and compare Facebook's API documentation to your mock API, and have a discussion in your group about what is different and what is the same?
```
Comment (benkahle) [inline lessons/06-apis-debugging/INCLASS.md]: Facebook has (at least) 3 distinct APIs, I might specify one specifically and provide a link to the documentation (https://developers.facebook.com/docs/graph-api/reference, perhaps). So folks know what they should be comparing their own design against.

File: lessons/06-apis-debugging/README.md [hunk 2]
```diff
@@ -1,9 +1,13 @@
 #Lesson 6 - APIs and Debugging Strategies
+##Preclass 
+Please read the following Readme and some of the following links on [Oauth](http://www.cubrid.org/blog/dev-platform/dancing-with-oauth-understanding-how-authorization-works/) and [REST](http://www.infoq.com/articles/rest-introduction)
+
 ##APIs: Your *real* on-ramp to the information superhighway
+*Author: Evan Simpson 2015*
```
Comment (benkahle) [inline lessons/06-apis-debugging/README.md]: Unless you're citing previous authors for a specific reason (or doing it consistently), it's probably cleaner to leave it out.

File: lessons/06-apis-debugging/README.md [hunk 3]
```diff
@@ -52,8 +56,18 @@ Access all your files stored online! Put them up, take them down, share them wit
 
 >**Documentation:** https://www.dropbox.com/developers
 
-###Netflix
-This one should not be on the list of APIs. Sadly, they closed their public API last November. Let's all take a moment to boo Netflix. Booooooooo.
+###US Gov
+A lot of government agencies have created APIs to access public data. You can see a list [here](git@github.com:olinjs/olinjs.git).
```
Comment (benkahle) [inline lessons/06-apis-debugging/README.md]: That URL is not correct.

File: lessons/06-apis-debugging/README.md [hunk 4]
```diff
@@ -52,8 +56,18 @@ Access all your files stored online! Put them up, take them down, share them wit
...
+
+###CURL
+When first accessing an API (especially one with poor documentation), you don't want to have to write a lot of code just to view the ouput of the request. You can use the command line tool CURL to quickly make requests.
```
Comment (benkahle) [inline lessons/06-apis-debugging/README.md]: -output

File: lessons/06-apis-debugging/passportExample/passport_example.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+});
+
+app.get('/auth/facebook', passport.authenticate('facebook'));
+
+app.get('/auth/facebook/callback',
+  passport.authenticate('facebook', { successRedirect: '/',
+                                      failureRedirect: '/login' }));
+
+app.get('/user', ensureAuthenticated, function(req, res) {
+	res.send(req.user);
```
Comment (benkahle) [inline lessons/06-apis-debugging/passportExample/passport_example.js]: keep indentation consistent

File: lessons/06-apis-debugging/passportExample/passport_example.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+})
+
+app.listen(3000);
+console.log('Listening on Port 3000');
+
+function ensureAuthenticated(req, res, next) {
+  if (req.isAuthenticated()) { return next(); }
+    res.send(401)
```
Comment (benkahle) [inline lessons/06-apis-debugging/passportExample/passport_example.js]: consistent use of semicolons

File: lessons/06-apis-debugging/sessionExample/index.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+			}
+			res.send(html);
+		});
+	}
+});
+
+app.post("/login", function(req, res) {
+	var sess = req.session;
+	sess.username = req.body.name;
+	res.redirect('..');
```
Comment (benkahle) [inline lessons/06-apis-debugging/sessionExample/index.js]: I would make this an explicit "/"

File: lessons/06-apis-debugging/README.md (hunk 3 above)
Comment (EvanSimpson) [inline lessons/06-apis-debugging/README.md]: https://www.data.gov/

Comment (rvanderheyde) [inline lessons/06-apis-debugging/README.md]: Oops

**Pull request closed without merging**